src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  todos: Todo[]
+  setTodos: (data: Todo[]) => void
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  todos,
+  setTodos,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [isEditing]);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleDoubleClick = () => setIsEditing(true);
+
+  const updateTitle = () => {
+    setIsEditing(false);
+
+    if (title === todo.title) {
+      return undefined;
```

Comment:
by default, the function returns an undefined anyway, so there is no need to write it again

```suggestion
      return;
```